### ISSUE TITLE
Search for function definition redirects to signature file (`.fsi`) instead of implementation

### ISSUE THREAD

**Issue by 64J0** [ionide/ionide-vscode-fsharp#1966]
### Describe the bug

In VSCode, when I have both the signature and implementation files for the same functions and I want to go for the function implementation, it redirects me to the signature instead. It would be very nice if this behavior could be customized to redirect to the implementation instead (maybe some switch on the extension configurations).

### Steps to reproduce

1. Create both signature (`.fsi`) and implementation (`.fs`) files for some function;
2. Call the function in another implementation file. Hover over the function name and hit Ctrl + Mouse left click (to redirect to the function implementation);
3. Notice that you'll be redirected to the signature file instead of the implementation.

### Expected behaviour

Redirect to the implementation instead of the signature.

If you're going to keep the redirect to the signature by default, it would be nice to have some configuration in the extension settings to allow it to redirect to the implementation instead of signature. 

## Machine info
 - OS: Linux
 - .NET SDK version: 7.0.401
 - Ionide version: 7.16.1

**Comment by MangelMaxime** [comment]
Some languages like TypeScript are able to provide a choice when there are several candidates:

![CleanShot 2023-12-05 at 20 22 08@2x](https://github.com/ionide/ionide-vscode-fsharp/assets/4760796/6c2d5e22-e6ee-492e-86aa-a0c0302e8db3)


I wonder if we should do something similar, it could also make it easier to navigate between signature and implementation files in general.

**Comment by baronfel** [comment]
Yeah - we have several gaps in navigation/references/etc with regards to separating implementation, declaration, and definition. A lot of this should come down to locating and returning different locations for 

* textdocument/declaration
* textdocument/definition
* textdocument/implementation

but we'd need to do that and see how far that gets us in covering the UX gaps.

**Comment by TheAngryByrd** [comment]
I started some of that work in https://github.com/fsharp/FsAutoComplete/pull/1111
